"EASY" install entry points are generated

With https://github.com/datalad/datalad/pull/4695 (against master, `maint` is not effected) we switched to use entry points generated by pip under assumption that with recent versions of pip we would get entry points which would not use `from pkg_resources import load_entry_point`.

After that tests of master within https://github.com/datalad/datalad-extensions/ started to fail on our unittest check that there is no EASY in entry point script:

https://pipelines.actions.githubusercontent.com/2UPlDxaVvvbkeFX4btxWorCjpJvj40zvWY5ogH2yZibhOMcU7O/_apis/pipelines/1/runs/1215/signedlogcontent/18?urlExpires=2020-08-07T17%3A01%3A07.5945328Z&urlSigningMethod=HMACV1&urlSignature=Vt72Dj9EnZwCagKG%2BoSyMTMDg2VCIx2jvarg6WqHogw%3D

```
2020-08-07T02:39:55.6249423Z ======================================================================
2020-08-07T02:39:55.6249720Z FAIL: datalad.cmdline.tests.test_main.test_script_shims
2020-08-07T02:39:55.6250546Z ----------------------------------------------------------------------
2020-08-07T02:39:55.6250923Z Traceback (most recent call last):
2020-08-07T02:39:55.6251824Z   File "/opt/hostedtoolcache/Python/3.7.8/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
2020-08-07T02:39:55.6252252Z     self.test(*self.arg)
2020-08-07T02:39:55.6253333Z   File "/opt/hostedtoolcache/Python/3.7.8/x64/lib/python3.7/site-packages/datalad/cmdline/tests/test_main.py", line 190, in test_script_shims
2020-08-07T02:39:55.6254187Z     assert_not_in('EASY', content) # NOTHING easy should be there
2020-08-07T02:39:55.6255195Z AssertionError: 'EASY' unexpectedly found in "#!/opt/hostedtoolcache/Python/3.7.8/x64/bin/python\n# EASY-INSTALL-ENTRY-SCRIPT: 'datalad==0.13.1','console_scripts','datalad'\n__requires__ = 'datalad==0.13.1'\nimport re\nimport sys\nfrom pkg_resources import load_entry_point\n\nif __name__ == '__main__':\n    sys.argv[0] = re.sub(r'(-script\\.pyw?|\\.exe)?$', '', sys.argv[0])\n    sys.exit(\n        load_entry_point('datalad==0.13.1', 'console_scripts', 'datalad')()\n    )\n"
2020-08-07T02:39:55.6255963Z 
2020-08-07T02:39:55.6256905Z ----------------------------------------------------------------------
2020-08-07T02:39:55.6258087Z Ran 1177 tests in 3464.796s
2020-08-07T02:39:55.6258412Z 
```

our datalad installation [code](https://github.com/datalad/datalad-extensions/blob/master/.github/workflows/build-git-annex-debianstandalone.yaml#L144) there first should upgrade pip and then proceed with installation of datalad:

```
          python -m pip install --upgrade pip
          pip install https://github.com/datalad/datalad/archive/${commitish}.zip
```

output in the log seems to support the fact that pip is first upgraded and then we proceed to the 2nd command to install datalad from github:

```
2020-08-07T01:41:51.7440572Z Collecting pip
2020-08-07T01:41:51.7797979Z   Downloading pip-20.2.1-py2.py3-none-any.whl (1.5 MB)
2020-08-07T01:41:51.9460441Z Installing collected packages: pip
2020-08-07T01:41:51.9463847Z   Attempting uninstall: pip
2020-08-07T01:41:51.9471333Z     Found existing installation: pip 20.2
2020-08-07T01:41:52.0977071Z     Uninstalling pip-20.2:
2020-08-07T01:41:52.1205880Z       Successfully uninstalled pip-20.2
2020-08-07T01:41:52.9926046Z Successfully installed pip-20.2.1
2020-08-07T01:41:53.7170964Z Collecting https://github.com/datalad/datalad/archive/master.zip
2020-08-07T01:41:53.8539923Z   Downloading https://github.com/datalad/datalad/archive/master.zip (1.8 MB)
```

but may be somehow because all commands come from the same "code block" 2nd pip invocation does not use freshly installed pip or smth like that?

any ideas @jwodder on what might be happening, i.e. why entry point shim is the EASY-INSTALL one?

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,3 @@
 [build-system]
 # wheel to get more lightweight (not EASY-INSTALL) entry-points
-requires = ["wheel"]
+requires = ["setuptools>=40.8.0", "wheel"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,0 +1,3 @@
+[build-system]
+# wheel to get more lightweight (not EASY-INSTALL) entry-points
+requires = ["wheel"]

